common/utils/serialise-url.ts

```diff
@@ -0,0 +1,27 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type QueryParams = { [key: string]: any };
+export type UrlParams = { [key: string]: string };
+
+export function parseUrlParams(
+  query: Readonly<QueryParams>,
+  paramsToConvert: string[]
+): UrlParams {
+  const newParams = { ...query }; // clone variable as it modifies original
+  // query contains global context, being specific on parsing
+  const itemToConvert = Object.keys(newParams).filter(param => {
+    if (paramsToConvert.includes(param)) {
+      return true;
+    }
+    return false;
+  });
+
+  // convert the items to string else return back
+  itemToConvert.forEach(key => {
```

Comment:
The reason this is mutating the original is because we're mutating an existing object.

A more safe way to do this would be:
```ts
const parsedQuery = Object.entries(query).reduce((acc, [key, val]) => {
  return {
    ...acc,
    [key]: parseQueryParam(val)
  }
}, {});
```

Why do we want to send a list of params to parse? If we don't, we can 100% guarantee that you will get a `{ [key: string]: string }` type back.

Reply:
Nice, yeah the solution is cleaner and safer, yeah my issue is query also contains these properties 

toggles
globalAlert
popupDialog
openingTimes
memoizedPrismic
query
workType

which are not really what i want to parse or modify. 

Should we just still allow this?